datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -164,6 +206,12 @@ impl<F: FileOpener> FileStream<F> {
                     partition_values,
                 } => match ready!(future.poll_unpin(cx)) {
                     Ok(reader) => {
+                        if let Some(instant) =
```

Comment:
I was wondering if it would be possible to use a `ScopedTimerGuard` rather than recording the starting instant yousefl

https://docs.rs/datafusion/10.0.0/datafusion/physical_plan/metrics/struct.ScopedTimerGuard.html

Reply:
looks 
```rust
pub struct ScopedTimerGuard<'a> {
    inner: &'a Time,
    start: Option<Instant>,
}
```
have a lifetime need lot code change🤔 and we need three kinds timer , in this api i think it only calculate a total one

Reply:
looks 
```rust
pub struct ScopedTimerGuard<'a> {
    inner: &'a Time,
    start: Option<Instant>,
}
```
have a lifetime need lot code change🤔 and we need three kinds timer , in this api i think it only calculate a total one

Reply:
Yeah you are right the lifetime stuff makes it a mess